### ISSUE TITLE
[iOS] Problem with snapToItem and lockScrollWhileSnapping=true

### ISSUE THREAD

**Issue by KarlosQ** [archriss/react-native-snap-carousel#177]
Hi!

If you try to do snapToItem the scroll goes back to the next item instead of going to index sent.

```
<Carousel
            ...
            enableMomentum={false}
            lockScrollWhileSnapping={true}
 />
```

If I comment 591 line to 595 it works
```
     // Snap as soon as next or previous item becomes active on iOS (buggy on Android)
    // WARNING: '_onScrollEnd()' is going to be called twice in a row
    /* if (this._canLockScroll() && IS_IOS &&
         (nextActiveItem === this._activeItem - 1 || nextActiveItem === this._activeItem + 1) &&
         (this._itemToSnapTo !== this._activeItem - 1 && this._itemToSnapTo !== this._activeItem + 1)) {
        this._onScrollEnd && this._onScrollEnd(event);
     }*/
```

React native: 0.48.4
RN-SC: 3.3.3 / 3.3.2 

Thanks for the support!

**Comment by bd-arc** [comment]
Hi @KarlosQ,

Well, good catch! I will need to either get rid of this feature or add yet another flag...

**Comment by bd-arc** [comment]
I've decided to remove the code since the associated feature wasn't essential but was really error-prone. The fix has been published in version `3.3.4` ;-)
